src/compiler/checker.ts

```diff
@@ -5665,6 +5665,12 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             if (getSymbolIfSameReference(exported, symbol)) {
                 return exported;
             }
+            if (symbol.flags & SymbolFlags.TypeAlias && exported.declarations?.find(isTypeAlias)) {

```

Comment:
I think we canonically use our own `find` helper here to avoid the `this` penalty; might be important if this is hot code.

Reply:
I have literally copied this bit from `getDeclaredTypeOfTypeAlias` :P
https://github.dev/microsoft/TypeScript/blob/cf8763b6c3193deefd095c86ce79921f23de307b/src/compiler/checker.ts#L12436

Should I switch both call sites to ur custom `find`?

Reply:
Hah, then nevermind.

Reply:
>  to avoid the `this` penalty
The original reason was that `find` wasn't defined in old runtimes. What is the `this` penalty? Maybe that's a good replacement reason.

Reply:
IIRC functions passed to `find` and other helpers like it are bound such that `this` in them points to the array.